packages/ragbits-core/src/ragbits/core/vector_stores/qdrant.py

```diff
@@ -168,6 +184,10 @@ async def list(  # type: ignore
         Raises:
             MetadataNotFoundError: If the metadata is not found.
         """
+        collection_exists = await self._client.collection_exists(collection_name=self._index_name)
```

Comment:
When can this happen? Shouldn't this result in an exception?

Reply:
This can happen when we try to list elements from the Qdrant collection that has not yet had anything saved to it.

Every time we are ingesting new documents we need to check whether any elements from these documents already exist in the vector store. This procedure checks this every time - even when the vector store is empty and we want to do the ingestion for the first time.

Originally, without this check, Qdrant raised a ValueError saying:
`Collection {collection_name} not found`
Now, if the collection doesn't exist, it just returns an empty list.